### PR TITLE
feat(modules and services) - task 05

### PR DESCRIPTION
**Pull request by kirillparamonov**
Task description: https://epam-my.sharepoint.com/:b:/p/nikita_reshetnikov/ERJrofkt1-1KgbJXbbumMaUBMgEfsuAB3ZsGWuyBKHhr1g?e=mvbmw1

### CODE REVIEW

File: src/app/services/course.service.ts [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
 
-  public verifyCourseBorder(courseCreationDate: Date): string {
-    const currentDate = moment();
-    const formattedCreationDate = moment(new Date(courseCreationDate));
-    const freshPeriodValue = moment().subtract(14, 'days');
-    const isCourseCreated = moment(currentDate).isAfter(formattedCreationDate);
-    const isCourseFresh = moment(formattedCreationDate).isSameOrAfter(freshPeriodValue);
-    const isCourseUpcoming = moment(formattedCreationDate).isAfter(currentDate);
+  public createCourse(courseItem: ICourse): void {
+    this.coursesList.push(courseItem);
```
Comment (EugeneCh) [inline src/app/services/course.service.ts]: should be new array like this.coursesList = [...this.courseList, courseItem], cause with onPush strategy this won't change the view

File: src/app/services/course.service.ts [hunk 2]
```diff
@@ -1,35 +1,32 @@
...
-    }
+  public getCourseById(id: number): ICourse {
+    return this.coursesList.find(course => course.id === id);
+  }
 
-    if (isCourseUpcoming) {
-      return '2px solid #30b6dd';
-    }
+  public updateCourse(updatedCourse: ICourse): void {
+    const coursesList = [...this.coursesList];
```
Comment (EugeneCh) [inline src/app/services/course.service.ts]: probably redundant, filter method already creates new array

File: src/app/services/course.service.ts [hunk 3]
```diff
@@ -1,35 +1,32 @@
...
+    this.coursesList = [...coursesList.filter(course => course.id !== updatedCourse.id), updatedCourse];
+  }
 
-    return '';
+  public removeCourseById(id: number): void {
+    const coursesList = [...this.coursesList];
```
Comment (EugeneCh) [inline src/app/services/course.service.ts]: same here

File: src/app/services/course.service.ts (hunk 1 above)
Comment (kirillparamonov) [reply src/app/services/course.service.ts]: Done, thanks!

File: src/app/services/course.service.ts (hunk 2 above)
Comment (kirillparamonov) [reply src/app/services/course.service.ts]: removed

File: src/app/services/course.service.ts (hunk 3 above)
Comment (kirillparamonov) [reply src/app/services/course.service.ts]: removed

**Pull request merged**